cloud/scope/vpc_cluster.go

```diff
@@ -1646,3 +1774,366 @@ func (s *VPCClusterScope) createSecurityGroupRuleRemote(remote infrav1beta2.VPCS
+				requeue = true
+			}
+			continue
+		}
+
+		// Otherwise, create the Load Balancer.
+		err = s.createLoadBalancer(loadBalancer)
+		if err != nil {
+			return false, err
+		}
+		// Assume a new Load Balancer will not be ready immediately, due to the complexity and time it takes.
+		requeue = true
+	}
+	return requeue, nil
+}
+
+// isLoadBalancerReady checks the state of a Load Balancer.
+// If state is active, true is returned, in all other cases, it returns false.
+// NOTE(cjschaef): May wish to extend this function to check all Load Balancer details (pools, listeners, etc.) as part of a Load Balancer being ready.
+func (s *VPCClusterScope) isLoadBalancerReady(status infrav1beta2.VPCLoadBalancerState) bool {
+	switch status {
+	case infrav1beta2.VPCLoadBalancerStateActive:
+		s.V(3).Info("load balancer is in active state")
+		return true
+	case infrav1beta2.VPCLoadBalancerStateCreatePending:
+		s.V(3).Info("load balancer is in create pending state")
+	default:
+		s.V(3).Info("load balancer is in unexpected state", "loadBalancerStatus", status)
+	}
+	return false
+}
+
+// getLoadBalancer attempts to retrieve the Load Balancer, otherwise returns nil if it doesn't exist.
+func (s *VPCClusterScope) getLoadBalancer(lb infrav1beta2.VPCLoadBalancerSpec) (*infrav1beta2.VPCLoadBalancerStatus, error) {
+	var loadBalancer *vpcv1.LoadBalancer
+	var err error
+	if lb.ID != nil {
+		var detailedResponse *core.DetailedResponse
+		getLBOptions := &vpcv1.GetLoadBalancerOptions{
+			ID: lb.ID,
+		}
+		loadBalancer, detailedResponse, err = s.VPCClient.GetLoadBalancer(getLBOptions)
+		if detailedResponse != nil && detailedResponse.StatusCode == http.StatusNotFound {
+			return nil, nil
+		}
+	} else {
+		name := lb.Name
+		if name == "" {
+			name = *s.GetServiceName(infrav1beta2.ResourceTypeLoadBalancer)
+		}
+		loadBalancer, err = s.VPCClient.GetLoadBalancerByName(name)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("error attempting to retrieve load balancer: %w", err)
+	}
+	if loadBalancer == nil {
+		return nil, nil
+	}
+	return &infrav1beta2.VPCLoadBalancerStatus{
+		ID:       loadBalancer.ID,
+		State:    infrav1beta2.VPCLoadBalancerState(*loadBalancer.ProvisioningStatus),
+		Hostname: loadBalancer.Hostname,
+	}, nil
+}
+
+// createLoadBalancer creates a Load Balancer.
+func (s *VPCClusterScope) createLoadBalancer(loadBalancer infrav1beta2.VPCLoadBalancerSpec) error {
+	options := &vpcv1.CreateLoadBalancerOptions{}
+	resourceGroupID, err := s.GetResourceGroupID()
+	if err != nil {
+		return err
+	}
+	if resourceGroupID == "" {
+		return fmt.Errorf("error getting resource group id for resource group %v, id is empty", s.IBMVPCCluster.Spec.ResourceGroup)
+	}
+
+	isPublic := true
+	// Load Balancer is private if defined that way (defaults to Public)
+	if loadBalancer.Public != nil && !*loadBalancer.Public {
+		isPublic = false
+	}
+
+	options.SetIsPublic(isPublic)
+	options.SetName(loadBalancer.Name)
```

Comment:
What if loadbalancer.Name is empty?
In getLoadBalancer() function we have this condition
```
name := lb.Name
if name == "" {
	name = *s.GetServiceName(infrav1beta2.ResourceTypeLoadBalancer)
}
```
Should we do something similar here?

Reply:
I'm hesitant to create LB's with a default service name, especially if multiple LB's get defined with empty names.
Given the structure of the VPCLoadBalancerStatus, we rely on ID as a priority field. We can let VPC API generate a Name, and rely on the ID from Status.

The default service name was added by request, but I don't expect to use it for most use cases. This is likely to be refactored, likely not relying on a default name, if possible.